src/filters/std/array.rs

```diff
@@ -47,57 +58,130 @@ impl Filter for JoinFilter {
     }
 }
 
+fn nil_safe_compare(a: &Value, b: &Value) -> Option<cmp::Ordering> {
```

Comment:
Should we modify `Value`s comparator instead?

Reply:
This (and `nil_safe_casecmp`) are implemented as separate functions in Ruby's implementation, so I went the same route.

Reply:
Interesting.  I can go either way.